linkerd/meshtls/boring/src/creds/store.rs

```diff
@@ -0,0 +1,135 @@
+use boring::{
+    pkey::{PKey, Private},
+    ssl,
+    x509::{store::X509Store, X509StoreContext, X509},
+};
+use linkerd_error::Result;
+use linkerd_identity as id;
+use tokio::sync::watch;
+
+pub struct Store {
+    roots: X509Store,
+    key: PKey<Private>,
+    csr: Vec<u8>,
+    name: id::Name,
+    client_tx: watch::Sender<ssl::SslConnector>,
+    server_tx: watch::Sender<ssl::SslAcceptor>,
+}
+
+// === impl Store ===
+
+impl Store {
+    pub(super) fn new(
+        roots: X509Store,
+        key: PKey<Private>,
+        csr: &[u8],
+        name: id::Name,
+        client_tx: watch::Sender<ssl::SslConnector>,
+        server_tx: watch::Sender<ssl::SslAcceptor>,
+    ) -> Self {
+        Self {
+            roots,
+            key,
+            csr: csr.into(),
+            name,
+            client_tx,
+            server_tx,
+        }
+    }
+
+    fn cert_matches_name(&self, cert: &X509) -> bool {
+        for san in cert.subject_alt_names().into_iter().flatten() {
+            if let Some(n) = san.dnsname() {
+                if let Ok(name) = n.parse::<linkerd_dns_name::Name>() {
```

Comment:
presumably `boring` also guarantees that the values returned by `dnsname()` are DNS names...does this differ significantly from our own validation in `linkerd_dns_name`?

Reply:
they should be the same, but is it really better to panic if there's a mismatch? it seems sufficient to fail to validate that the cert matches the name.

Reply:
yeah, i think this is fine, i was mostly curious about whether there was a possible difference.